frontend/src/plugin/Plugins.tsx

```diff
@@ -23,21 +23,53 @@ export default function Plugins() {
 
     /**
      * Get the list of plugins,
-     *   download all the plugins,
+     *   download all the plugin source,
+     *   download all the plugin package.json files,
      *   execute the plugins,
      *   .initialize() plugins that register (not all do).
      */
     async function fetchAndExecute() {
-      const pluginsScriptPaths = (await fetch(`${helpers.getAppUrl()}plugins/paths`).then(resp =>
+      const pluginPaths = (await fetch(`${helpers.getAppUrl()}plugins/paths`).then(resp =>
         resp.json()
       )) as string[];
 
-      const sources = await Promise.all(
-        pluginsScriptPaths.map(path =>
+      const sourcesPromise = Promise.all(
+        pluginPaths.map(path =>
           fetch(`${helpers.getAppUrl()}${path}/main.js`).then(resp => resp.text())
         )
       );
 
+      // fetch the packages. But if there is a problem,
+      const packageInfosPromise = await Promise.all(
+        pluginPaths.map(path =>
+          fetch(`${helpers.getAppUrl()}${path}/package.json`).then(resp => {
+            if (!resp.ok) {
+              if (resp.status !== 404) {
+                return Promise.reject(resp);
+              }
+              {
+                console.warn(
+                  'Missing package.json. ' +
+                    `Please upgrade the plugin at ${path}` +
```

Comment:
We can refer to the plugins names given by the plugin endpoint without treating it as a path. So `Please upgrade the plugin ${path}` (and renaming path to `pluginName` or something) would make more sense to me.

Reply:
This was changed.